javaslang/src/main/java/javaslang/collection/TreeMap.java

```diff
@@ -859,8 +854,7 @@ public boolean containsKey(K key) {
 
     @Override
     public <K2, V2> TreeMap<K2, V2> flatMap(Comparator<? super K2> keyComparator,
-                                            BiFunction<? super K, ? super V, ? extends Iterable<Tuple2<K2, V2>>> mapper) {
-        Objects.requireNonNull(keyComparator, "keyComparator is null");
+            BiFunction<? super K, ? super V, ? extends Iterable<Tuple2<K2, V2>>> mapper) {
```

Comment:
The TreeMap.EntryComparator has two use-cases:

1. the natural key-comparator
2. a specific key-comparator

The two belong together here. The check for the _specfic_ case is now performed in the EntryComparator factory method, which is common for all callers.